### PR TITLE
HTTP endpoints cleanup

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Cleanup

**What does this PR do? Why is it needed?**

- Splits structs and conversion functions into three files: `state`, `block` and everything else
- Moves structs and conversion functions used only by the validator to the validator client's code
- Removes the validation package json tags
- Removes nil checks at the beginning of `FromConsensus` functions
- Unifies uint64->string conversions in `FromConsensus`by using `fmt.Sptrinf` everywhere
- Returns exact param names in handler error messages
- Renames our `http` package to `httputil`
- Updates how query and route params are handled
- Adds spans to custom Prysm endpoints
- Renames `Bls` to `BLS` in field names
- Moves these items from `beacon-chain/rpc/eth/shared` to dedicated packages:
  - `Uint256ToSSZBytes` to `encoding/bytesutil/integers.go`
  - `DecodeHexWithLength` and `DecodeHexWithMaxLength` to `encoding/bytesutil.hex.go`
  - `DecodeError` and `IndexedVerificationFailureError` to `api/server/error.go`

**Which issues(s) does this PR fix?**

Addresses several points from #12958

### CODE REVIEW

File: beacon-chain/rpc/prysm/node/handlers.go [hunk 1]
```diff
@@ -86,23 +93,26 @@ func (s *Server) AddTrustedPeer(w http.ResponseWriter, r *http.Request) {
 		s.PeersFetcher.Peers().Add(nil, info.ID, info.Addrs[0], direction)
 	}
 
-	var peers []peer.ID
-	peers = append(peers, info.ID)
-	s.PeersFetcher.Peers().SetTrustedPeers(peers)
+	var ids []peer.ID
```
Comment (rkapka) [inline beacon-chain/rpc/prysm/node/handlers.go:96]: collision with package name

File: beacon-chain/rpc/prysm/node/handlers.go [hunk 2]
```diff
@@ -112,9 +122,9 @@ func (s *Server) RemoveTrustedPeer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var peers []peer.ID
-	peers = append(peers, peerId)
-	s.PeersFetcher.Peers().DeleteTrustedPeers(peers)
+	var ids []peer.ID
```
Comment (rkapka) [inline beacon-chain/rpc/prysm/node/handlers.go:125]: collision with package name

File: beacon-chain/rpc/eth/shared/request.go [hunk 3]
```diff
@@ -8,66 +8,95 @@ import (
...
-
+	"github.com/gorilla/mux"
 	"github.com/prysmaticlabs/prysm/v4/beacon-chain/blockchain"
 	"github.com/prysmaticlabs/prysm/v4/beacon-chain/sync"
-	http2 "github.com/prysmaticlabs/prysm/v4/network/http"
+	"github.com/prysmaticlabs/prysm/v4/network/httputil"
 )
 
-func UintFromQuery(w http.ResponseWriter, r *http.Request, name string) (bool, string, uint64) {
+func UintFromQuery(w http.ResponseWriter, r *http.Request, name string, allowEmpty bool) (bool, string, uint64) {
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/shared/request.go:17]: what about allowEmpty -> required

File: beacon-chain/rpc/eth/beacon/handlers_state.go [hunk 4]
```diff
@@ -34,20 +34,20 @@ func (s *Server) GetStateRoot(w http.ResponseWriter, r *http.Request) {
 
 	stateId := mux.Vars(r)["state_id"]
 	if stateId == "" {
-		http2.HandleError(w, "state_id is required in URL params", http.StatusBadRequest)
+		httputil.HandleError(w, "state_id is required in URL params", http.StatusBadRequest)
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/beacon/handlers_state.go:37]: similar to what we did with blocker, what if we did the same thing for the other endpoints?
extract the core logic out and return response/rpcErr so we don't need a static analyzer for all these httputil.HandleError 🤔

File: beacon-chain/rpc/eth/shared/conversions_state.go [hunk 5]
```diff
@@ -0,0 +1,592 @@
...
+	rm := make([]string, len(srcrm))
+	for i, m := range srcrm {
+		rm[i] = hexutil.Encode(m)
+	}
+	srcslashings := st.Slashings()
+	slashings := make([]string, len(srcslashings))
+	for i, s := range srcslashings {
+		slashings[i] = fmt.Sprintf("%d", s)
+	}
+	srcPrevAtts, err := st.PreviousEpochAttestations()
```
Comment (saolyn) [inline beacon-chain/rpc/eth/shared/conversions_state.go:55]: This is a bit of a nit, but some of these are camel case while others are all lower case

File: beacon-chain/rpc/eth/shared/structs.go [hunk 6]
```diff
@@ -1,19 +1,34 @@
...
-	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/pkg/errors"
-
-	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
-	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
-	"github.com/prysmaticlabs/prysm/v4/consensus-types/validator"
-	eth "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
 )
 
+var errPayloadHeaderNotFound = errors.New("expected payload header not found")
```
Comment (saolyn) [inline beacon-chain/rpc/eth/shared/structs.go:7]: This error is only used in `beacon-chain/rpc/eth/shared/conversions_state.go` might be better to move it in there

File: testing/endtoend/BUILD.bazel [hunk 7]
```diff
@@ -173,8 +173,8 @@ go_test(
         "@web3signer",
     ],
     eth_network = "mainnet",
-    flaky = True,
-    shard_count = 2,
+    # flaky = True,
+    # shard_count = 2,
```
[the remark this answers is not in the document]
Comment (rkapka) [reply testing/endtoend/BUILD.bazel:177]: oops

File: beacon-chain/rpc/eth/beacon/handlers_state.go (hunk 4 above)
Comment (rkapka) [reply beacon-chain/rpc/eth/beacon/handlers_state.go:37]: We would still need to remember about typing `return` after calling these functions

**Review by james-prysm — APPROVED** [review]
thanks for the cleanup there a few more areas we can talk about but keep that separated from this PR.

**Pull request merged**